### ISSUE TITLE
Implement MASS objective for denoising autoencoder

### ISSUE THREAD

**Issue by Waino** [Helsinki-NLP/mammoth#15]
According to Baziotis et al [1], the MASS objective [2] is superior to BART, for training on monolingual data using the denoising sequence auto-encoder approach.

BART (alreday implemented in mammoth) is easier to implement though, as it fits with the transform framework. MASS on the other hand requires:

- [x] Two separate target language sequences: decoder input and output. (x_5 is replaced by a mask symbol in the input)
- [ ] Token-level masking of the loss. Only the three unmasked tokens x_3--x_5 contribute to the loss. The other tokens can be generated (as it is easier than not doing so), but their contribution to the loss should be zeroed out.

Both MASS and BART use sampled noise, and must be applied online during training, as opposed to cleaning preprocessing (#13) that can also be applied offline.

Note that MASS is tricky to implement, while BART we already have. It may not be worth the effort to implement this.

![image](https://github.com/Helsinki-NLP/mammoth/assets/1087483/acf91d31-49a0-4365-9df3-46da1c62e79f)

[1] Baziotis et. al (2023) "When Does Monolingual Data Help Multilingual Translation: The Role of Domain and Model Scale" https://arxiv.org/abs/2305.14124
[2] Song et. al (2019) "MASS: Masked Sequence to Sequence Pre-training for Language Generation" https://arxiv.org/abs/1905.02450
